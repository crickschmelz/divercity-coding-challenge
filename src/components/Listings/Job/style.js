import styled from "styled-components";

export const Div = styled.div`
  margin-left: 10px;
`;

export const ApplyButton = styled.button`
  color: #ffffff;
  background-color: #3197e4;
  border: none;
  font-weight: 800;
  font-size: 14px;
  width: 100px;
  padding: 10px;
  margin: 10px 0 20px 0;
  border-radius: 30px;
  cursor: pointer;
`;

export const Form = styled.form`
  display: grid;
  justify-content: center;
`;

export const Modal = styled.div`
  position: absolute;
  z-index: 1;
  height: 100%;
  width: 100%;
  background-color: #f8e1e9;
  opacity: 0.95;
`;

export const Input = styled.input`
  display: block;
  padding: 2%;
  margin: 2%;
  color: palevioletred;
  background: papayawhip;
  border: none;
  border-radius: 3px;
`;

export const Header = styled.h3`
  color: #072559;
  border-top: 1px solid black;
  padding-top: 3%;
`;
