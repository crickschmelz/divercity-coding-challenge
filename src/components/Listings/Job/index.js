import React from "react";
import PropTypes from "prop-types";
import { Header, ApplyButton, Div } from "./style";

const Job = (props) => {
  Job.propTypes = {
    openApplication: PropTypes.func,
    title: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string,
    company: PropTypes.string
  };

  const { openApplication, title, description, location, company } = props;

  function createMarkup() {
    return { __html: description };
  }

  return (
    <Div>
      <Header>
        {company} - {title}
      </Header>
      <h4>{location}</h4>
      <div dangerouslySetInnerHTML={createMarkup()} />
      <ApplyButton onClick={openApplication}>Apply</ApplyButton>
    </Div>
  );
};

export default Job;
