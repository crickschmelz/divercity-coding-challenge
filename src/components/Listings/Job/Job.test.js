import React from "react";
import Job from "./";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Job />).toJSON();
  expect(tree).toMatchSnapshot();
});
