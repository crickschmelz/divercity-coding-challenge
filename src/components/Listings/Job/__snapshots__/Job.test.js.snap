// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm bQKYXJ"
>
  <h3
    className="sc-AxgMl endhLy"
  >
     - 
  </h3>
  <h4 />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <button
    className="sc-AxirZ dLeGRp"
  >
    Apply
  </button>
</div>
`;
